be/src/env/cloud_storage_client.h

```diff
@@ -0,0 +1,53 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+#pragma once
+
+#include <string>
+
+#include "common/status.h"
+
+namespace starrocks {
+
+// base class for cloud storage client
+class CloudStorageClient {
+public:
+    CloudStorageClient() = default;
+    virtual ~CloudStorageClient() = default;
+
+    /*
+     *  Bucket Operation
+     */
+    virtual Status create_bucket(const std::string& bucket_name) = 0;
+
+    virtual Status delete_bucket(const std::string& bucket_name) = 0;
+
+    /*
+     *  Object Operation
+     */
+    virtual Status put_object(const std::string& bucket_name, const std::string& object_key,
```

Comment:
Nit: 
Declare the interface as 
```
    virtual StatusOr<io::OutputStream> put_object(const std::string& bucket_name, const std::string& object_key);
```
and writing an object through io::OutputStream.

Reply:
Most interfaces are used by Cloud Team. Maybe they have concern. Better to keep it at first.